tsdb/engine/tsm1/engine.go

```diff
@@ -390,12 +390,7 @@ func (e *Engine) addToIndexFromKey(shardID uint64, key string, fieldType influxq
 	m := index.CreateMeasurementIndexIfNotExists(measurement)
 	m.SetFieldName(field)
 
-	mf := e.measurementFields[measurement]
-	if mf == nil {
-		mf = tsdb.NewMeasurementFields()
-		e.measurementFields[measurement] = mf
-	}
-
+	mf := e.MeasurementFields(measurement)
```

Comment:
@jwilder I just looked at this and realised it's the `Cache`'s `RLock` that's taken in `LoadMetaDataIndex`, not the `Engine`'s.

So we're good to go?